packages/ra-ui-materialui/src/field/BooleanField.spec.tsx

```diff
@@ -150,6 +150,22 @@ describe('<BooleanField />', () => {
         }
     );
 
+    it.each([null, undefined])(
+        'should display emptyText ReactNode prop if provided for %s value',
+        published => {
+            const emptyTextNode = <b>Custom Empty Text</b>;
+            render(
+                <BooleanField
+                    {...defaultProps}
+                    record={{ id: 123, published }}
+                    emptyText={emptyTextNode}
+                />
+            );
+            const renderedEmptyText = screen.getByText('Custom Empty Text');
+            expect(renderedEmptyText).toBeTruthy();
```

Comment:
`getByXXX` throws if the searched element is not found. This `expect` doesn't really make sense. So either use `queryByYYY` or remove the `expect`. 

See https://testing-library.com/docs/react-testing-library/cheatsheet/#queries

Same for other tests.